File: rb-appscript/tags/rb-appscript-0.5.0/sample/Make_Mail_message.rb
#!/usr/bin/env ruby

# Compose an outgoing message in Apple's Mail.app.

# Note: if using the appscript gem, rubygems must be required first:
begin; require 'rubygems'; rescue LoadError; end

require "appscript"
include Appscript

def make_message(addresses, subject, content, show_window=false)
	# Make an outgoing message in Mail.
	# 	addresses : list of unicode -- a list of email addresses
	# 	subject : unicode -- the message subject
	# 	content : unicode -- the message content
	# 	show_window : Boolean -- show message window in Mail
	# Result : reference -- reference to the new outgoing message
	mail = app('Mail')
	msg = mail.make(
			:new => :outgoing_message, 
			:with_properties => {:visible => show_window})
	addresses.each do |an_address|
		msg.to_recipients.end.make(
				:new => :recipient, 
				:with_properties => {:address => an_address})
	end
	msg.subject.set(subject)
	msg.content.set(content)
	return msg
end

# test
p make_message(['joe@foo.com', 'jane@bar.net'], 'Hello World', 'Some body text.', true)

File: py-appscript/tags/py-appscript-0.21.1/appscript_3x/sample/appscript/Make_Mail_message.py
#!/usr/bin/env python3

# Compose an outgoing message in Apple's Mail.app.

from appscript import *

def makemessage(addresses, subject, content, showwindow=False):
    """Make an outgoing message in Mail.
        addresses : list of unicode -- a list of email addresses
        subject : unicode -- the message subject
        content : unicode -- the message content
        showwindow : Boolean -- show message window in Mail
        Result : reference -- reference to the new outgoing message
   	"""
    mail = app('Mail')
    msg = mail.make(
            new=k.outgoing_message, 
            with_properties={k.visible: showwindow})
    for anaddress in addresses:
        msg.to_recipients.end.make(
                new=k.recipient, 
                with_properties={k.address: anaddress})
    msg.subject.set(subject)
    msg.content.set(content)
    return msg


# test
print(makemessage(['joe@example.com', 'jane@example.net'],
        'Hello World', 'Some body text.', True))